scripts/metalk8s_cli/mixins/salt.py

```diff
@@ -0,0 +1,119 @@
+import json
+import shlex
+import subprocess
+
+import six
+
+from metalk8s_cli.exceptions import CommandError, CommandInitError
+
+
+class SaltCommandMixin(object):
+    def __init__(self, args):
+        super(SaltCommandMixin, self).__init__(args)
+
+    def run_salt_minion(
+        self, command, local=False, outputter=None, color=True,
+        saltenv=None, pillar=None,
+    ):
+        full_cmd = ['salt-call']
+
+        if local:
+            full_cmd.append('--local')
+        if outputter:
+            full_cmd.extend(['--out', outputter])
+        if color:
+            full_cmd.append('--force-color')
+
+        full_cmd.extend(self._build_args(command))
+
+        if saltenv is not None:
+            full_cmd.append('saltenv={}'.format(saltenv))
+        if pillar is not None:
+            full_cmd.append("pillar='{}'".format(json.dumps(pillar)))
+
+        try:
+            return subprocess.check_output(full_cmd, stderr=subprocess.STDOUT)
```

Comment:
Why do we, at several places, redirect `stderr` into `stdout`?

Why not using `subprocess.Popen` and then use `communicate` to get the output (`retcode` will be in the object returned by `Popen`)?

Reply:
The goal was to print `stdout`/`stderr` if the user requests `--verbose` mode only, but as @TeddyAndrieux suggested, we should not mix them if we want to parse the `stdout` result.
I moved to `Popen.communicate` since then, push is coming soon